skbeam/core/accumulators/binned_statistic.py

```diff
@@ -130,7 +136,13 @@ def __init__(self, sample, statistic='mean',
         # Compute the bin number each sample falls into.
         Ncount = {}
         for i in np.arange(self.D):
-            Ncount[i] = np.digitize(sample[:, i], self.edges[i])
+            # Apply mask in a non-ideal way by setting value outside range.
+            # Would be better to do this using bincount "weights", perhaps.
+            thissample = sample[:, i]
+            thismask = mask[:, i]
+            thissample[thismask == 0] = (self.edges[i][0] -
+                                         0.01 * (1+np.fabs(self.edges[i][0])))
+            Ncount[i] = np.digitize(thissample, self.edges[i])
```

Comment:
before version 1.10.0 (just last year), numpy had an excruciatingly slow version of np.digitize
(was fixed here https://github.com/numpy/numpy/pull/5101). I don't think any python user will encounter this again, but could we add some note regarding this somewhere just in case?
See release notes : https://github.com/numpy/numpy/blob/master/doc/release/1.10.0-notes.rst

I suggest in the function doc:
Notes
--------
If using numpy versions < 1.10.0, you may notice significantly slow behavior. This has to do with digitize, which was optimized 1.10.0 onwards.